lms/djangoapps/email_marketing/signals.py

```diff
@@ -139,6 +140,25 @@ def email_marketing_register_user(sender, user, registration,
                       site=_get_current_site(), new_user=True)
 
 
+@receiver(REGISTER_ENTERPRISE_USER)
+def email_marketing_enterprise_user(sender, user, **kwargs):  # pylint: disable=unused-argument
```

Comment:
Move this to `openedx/features/enterprise_support/signals.py`. Use the `post_save` signal fired by the `EntepriseCustomerUser` model instead of the custom one you created.